huaweicloud/resource_huaweicloud_compute_instance.go

```diff
@@ -681,27 +681,45 @@ func resourceComputeInstanceV2Create(ctx context.Context, d *schema.ResourceData
 		}
 	}
 
-	if err := resourceComputeInstanceV2Read(ctx, d, meta); err != nil {
-		return err
-	}
+	// get the original value of source_dest_check in script
+	originalNetworks := d.Get("network").([]interface{})
+	sourceDestChecks := make([]bool, len(originalNetworks))
+	var flag bool
 
-	// note: as "network.port" is optional, we shoud use it after resourceComputeInstanceV2Read
-	networks := d.Get("network").([]interface{})
-	for _, v := range networks {
+	for i, v := range originalNetworks {
 		nic := v.(map[string]interface{})
-		nicPort := nic["port"].(string)
-		if nicPort == "" {
-			continue
+		sourceDestChecks[i] = nic["source_dest_check"].(bool)
+		if !sourceDestChecks[i] {
+			flag = true
 		}
+	}
 
-		if sourceDestCheck := nic["source_dest_check"].(bool); !sourceDestCheck {
-			if err := disableSourceDestCheck(nicClient, nicPort); err != nil {
-				return diag.Errorf("error disable source dest check on port(%s) of instance(%s) failed: %s", nicPort, d.Id(), err)
+	if flag {
+		// Get the instance network and address information
+		server, err := cloudservers.Get(ecsClient, d.Id()).Extract()
+		if err != nil {
+			return diag.Errorf("error retrieving compute instance: %s", d.Id())
+		}
+		networks, err := flattenInstanceNetworks(d, meta, server)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+
+		for i, nic := range networks {
+			nicPort := nic["port"].(string)
+			if nicPort == "" {
+				continue
+			}
+
+			if !sourceDestChecks[i] {
+				if err := disableSourceDestCheck(nicClient, nicPort); err != nil {
+					return diag.Errorf("error disable source dest check on port(%s) of instance(%s) failed: %s", nicPort, d.Id(), err)
```

Comment:
maybe `error disable source dest check on port(%s) of instance(%s)` or `disable source dest check on port(%s) of instance(%s) failed` is beter.

Reply:
fixed